src/classes/emitter.ts

```diff
@@ -41,33 +43,29 @@ export default class Emitter<Events extends EventMap = EventMap> {
 		this.prototype.off = this.prototype.removeListener;
 	}
 
-	// @ts-expect-error this is correct since we're overwriting with specific args.
 	on<T extends keyof Events>(
 		eventName: T,
 		listener: (this: this, ...args: Events[T]) => void,
-	): this;
-	on(eventName: EventName, listener: Listener): this {
-		const listeners = this.#events.get(eventName);
+	): this {
```

Comment:
we can use only the typed function definition.
But we have to change some small things